numpy_financial/_financial.py

```diff
@@ -571,12 +572,18 @@ def pv(rate, nper, pmt, fv=0, when='end'):
 #  p*((r + 1)^n - 1)*(r*w + 1)/r^2 + n*p*(r + 1)^(n - 1)*(r*w + 1)/r +
 #  p*((r + 1)^n - 1)*w/r)
 
+
 def _g_div_gp(r, n, p, x, y, w):
+    # Evaluate g(r_n)/g'(r_n), where g =
+    # fv + pv*(1+rate)**nper + pmt*(1+rate*when)/rate * ((1+rate)**nper - 1)
     t1 = (r+1)**n
     t2 = (r+1)**(n-1)
-    return ((y + t1*x + p*(t1 - 1)*(r*w + 1)/r) /
-                (n*t2*x - p*(t1 - 1)*(r*w + 1)/(r**2) + n*p*t2*(r*w + 1)/r +
-                 p*(t1 - 1)*w/r))
+    g = y + t1*x + p*(t1 - 1) * (r*w + 1) / r
+    gp = n*t2*x
```

Comment:
This line needs to have an opening parenthesis, e.g. `gp = (n*t2*x`, and the line that ends the expression gets the closing parenthesis.  Without those parentheses, the code simply assigns `n*t2*x` to `gp`, and the next three lines (which each happens to be a syntactically correct expression) are evaluated and then ignored.

Clearly the test coverage of this code is insufficient. 😮

Reply:
I've check this, the test calculates the result at the end of the period (i.e. w=0) so all the lines containing zero evaluate to 0 and the test passes. 

I'd prefer to get these moved across *as is* before making any behavioural changes, would it ok with you if I create an issue to increase the test coverage, using this as an example?

Reply:
I didn't mean to suggest that tests were needed in *this* pull request!  

> would it ok with you if I create an issue to increase the test coverage, using this as an example?


Yes, definitely.  We might as well keep track of the issue, even if we don't improve it right away.